src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -400,13 +408,21 @@ internal void ApplyProjectUpdateState(ProjectUpdateState projectUpdateState)
     {
         Contract.ThrowIfFalse(_gate.CurrentCount == 0);
 
-        // Now that we've removed the references from the sln, we can stop watching them.
+        // Remove file watchers for any references we're no longer watching.
         foreach (var reference in projectUpdateState.RemovedMetadataReferences)
-            FileWatchedReferenceFactory.StopWatchingReference(reference);
+            FileWatchedPortableExecutableReferenceFactory.StopWatchingReference(reference);
 
-        // Now that we've added the references to the sln, we can start watching them.
+        // Add file watchers for any references we are now watching.
         foreach (var reference in projectUpdateState.AddedMetadataReferences)
-            FileWatchedReferenceFactory.StartWatchingReference(reference, reference.FilePath!);
+            FileWatchedPortableExecutableReferenceFactory.StartWatchingReference(reference, reference.FilePath!);
+
+        // Remove file watchers for any references we're no longer watching.
```

Comment:
>references [](http://example.com/codeflow?start=40&length=10)

nit: consider "metadata references" and "analyzer references" in the comments to distinguish

Reply:
can add in followup.